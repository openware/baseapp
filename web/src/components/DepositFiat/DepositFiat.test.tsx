import { render, screen } from '@testing-library/react';
import React from 'react';
import { TestComponentWrapper } from 'src/lib/test';
import { DepositFiat } from './';

describe('DepositFiat', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = render(
            <TestComponentWrapper>
                <DepositFiat title={'Custom Title'} description={'Custom Description'} uid={'42389734'} />
            </TestComponentWrapper>,
        );
    });

    it('should contains title', () => {
        expect(screen.getByText('Custom Title')).toBeInTheDocument();
    });

    it('should contains right description', () => {
        expect(screen.getByText('Custom Description')).toBeInTheDocument();
    });

    it('should match snapshot', () => {
        expect(wrapper.container).toMatchSnapshot();
    });
});
