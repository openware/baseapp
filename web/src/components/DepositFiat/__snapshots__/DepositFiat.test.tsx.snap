// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DepositFiat should match snapshot 1`] = `
<div>
  <div
    class="cr-deposit-fiat"
  >
    <p
      class="cr-deposit-fiat__title"
    >
      Custom Title
    </p>
    <p
      class="cr-deposit-fiat__description"
    >
      Custom Description
    </p>
    <div
      class="cr-deposit-fiat-credentials"
    >
      <div
        class="cr-deposit-fiat-detail"
      >
        <p
          class="cr-deposit-fiat-detail__label"
        >
          Bank Name
          :
        </p>
        <p
          class="cr-deposit-fiat-detail__value"
        >
          Diamant Bank
        </p>
      </div>
      <div
        class="cr-deposit-fiat-detail"
      >
        <p
          class="cr-deposit-fiat-detail__label"
        >
          Account number
          :
        </p>
        <p
          class="cr-deposit-fiat-detail__value"
        >
          10120212
        </p>
      </div>
      <div
        class="cr-deposit-fiat-detail"
      >
        <p
          class="cr-deposit-fiat-detail__label"
        >
          Account name
          :
        </p>
        <p
          class="cr-deposit-fiat-detail__value"
        >
          name
        </p>
      </div>
      <div
        class="cr-deposit-fiat-detail"
      >
        <p
          class="cr-deposit-fiat-detail__label"
        >
          Phone Number
          :
        </p>
        <p
          class="cr-deposit-fiat-detail__value"
        >
          +3 8093 1212 12 12
        </p>
      </div>
      <div
        class="cr-deposit-fiat-detail"
      >
        <p
          class="cr-deposit-fiat-detail__label"
        >
          Your reference code
          :
        </p>
        <p
          class="cr-deposit-fiat-detail__value"
        >
          42389734
        </p>
      </div>
    </div>
  </div>
</div>
`;
